File: http-client/MCS.WatchTower.WebApi.DataTransferObjects/Responses/AppResponse.cs
namespace MCS.WatchTower.WebApi.DataTransferObjects.Responses;

public record AppResponse : IBaseDto
{
    public string Id { get; init; }
    public string HostName { get; init; }
    public string AppName { get; init; }
    public string Port { get; init; }

    public string Type { get; init; }
    public AppStatusResponse LatestStatus { get; init; }

    public string CronStartTime { get; init; }
    public string CronStopTime { get; init; }

    public string Version { get; init; }
    public ISet<string> Tags { get; init; }
}

File: frontend/src/app/features/apps/models/app-response.ts
import { BaseModel } from "../../../core/models/base-model";
import { AppStatusResponse } from "../../app-statuses/models/app-status-response";

export interface AppResponse extends BaseModel {
  hostName: string;
  appName: string;
  port: string;

  type: string;
  latestStatus: AppStatusResponse;

  cronStartTime: string;
  cronStopTime: string;

  version: string;
  tags: string[];
}
